querybook/server/lib/query_executor/clients/trino.py

```diff
@@ -61,14 +62,27 @@ def __init__(
         host = trino_conf.host
         port = 8080 if not trino_conf.port else trino_conf.port
 
-        connection = trino.dbapi.connect(
-            host=host,
-            port=port,
-            catalog=trino_conf.catalog,
-            schema=trino_conf.schema,
-            user=proxy_user or username,
-            http_scheme=trino_conf.protocol,
-        )
+        if username is not None and password is not None:
+            auth = trino.auth.BasicAuthentication(username, password)
+
+            connection = trino.dbapi.connect(
+                host=host,
+                port=port,
+                catalog=trino_conf.catalog,
+                schema=trino_conf.schema,
+                auth=auth,
+                user=proxy_user if proxy_user else username,
+                http_scheme=trino_conf.protocol,
+            )
+        else:
+            connection = trino.dbapi.connect(
+                host=host,
+                port=port,
+                catalog=trino_conf.catalog,
+                schema=trino_conf.schema,
+                user=proxy_user or username,
+                http_scheme=trino_conf.protocol,
+            )
```

Comment:
no need to duplicate this code
```suggestion
        auth = trino.constants.DEFAULT_AUTH
        if username is not None and password is not None:
            auth = trino.auth.BasicAuthentication(username, password)

        connection = trino.dbapi.connect(
            host=host,
            port=port,
            catalog=trino_conf.catalog,
            schema=trino_conf.schema,
            auth=auth,
            user=proxy_user if proxy_user else username,
            http_scheme=trino_conf.protocol,
        )
```

Reply:
Done